server/game/cards/20-HMW/NoSongSoSweet.js

```diff
@@ -0,0 +1,35 @@
+const DrawCard = require('../../drawcard.js');
+
+class NoSongSoSweet extends DrawCard {
+    setupCardAbilities(ability) {
+        this.action({
+            title: 'Add to challenge',
+            condition: () => this.game.isDuringChallenge({ challengeType: 'power' }),
+            target: {
+                cardCondition: card => card.location === 'play area' && !card.kneeled && card.controller === this.controller &&
+                                       card.isFaction('baratheon') && card.canParticipateInChallenge()
+            },
+            message: '{player} plays {source} to add {target} to the challenge',
+            handler: context => {
+                this.game.currentChallenge.addParticipantToSide(context.player, context.target);
+            }
+        });
+        
+        this.reaction({
+            location: 'discard pile',
+            when: {
+                onCardPlayed: event => event.card.hasTrait('Song') && event.card.controller === this.controller
+            },
+            ignoreEventCosts: true,
+            cost: ability.costs.payGold(1),
+            handler: () => {
+                this.game.addMessage('{0} returns {1} back to their hand', this.controller, this);
+                this.controller.moveCard(this, 'hand');
```

Comment:
This should use `context.player`, and should use the returnCardToHand game action. (And doesn't seem to match the text on ThronesDB?)

Reply:
fixed and i think it is in line with the thronesdb state.
quote from thronesdb:


Action: During a challenge, choose a standing character you control and have it participate in the challenge on your side.

Reaction: After you play a Song event, pay 1 gold to return No Song So Sweet from your discard pile to your hand.

Maybe your cache of thronesdb is out of date?